src/Evolution/Systems/Cce/OptionTags.hpp

```diff
@@ -320,6 +329,28 @@ struct H5WorldtubeBoundaryDataManager : db::SimpleTag {
   }
 };
 
+/// A tag that constructs a `KleinGordonWorldtubeDataManager` from options
+struct KleinGordonH5WorldtubeBoundaryDataManager : db::SimpleTag {
```

Comment:
Why not combine this tag with the existing one? It doesn't need any extra parameters. Then instead of having a separate option for `KleinGordonBoundaryDataFilename`, you change the `H5IsBondiData` tag to be an enum (which you'd have to add) like

```cpp
enum class H5DataType { CartesianMetric, Bondi, KleinGordon };
```

and just use that to decide which `WorldtubeDataManager` to construct.

Reply:
This manager is designed to handle only the Klein-Gordon tags (`Tags::klein_gordon_worldtube_boundary_tags`). The other tensor tags (`Tags::characteristic_worldtube_boundary_tags<Tags::BoundaryValue>`) are still managed by the current `H5WorldtubeBoundaryDataManager`. In other words, there will be two managers in the scalar-tensor CCE system, in order to deal with boundary data separately (`Tags::klein_gordon_worldtube_boundary_tags`+`Tags::characteristic_worldtube_boundary_tags<Tags::BoundaryValue>`).

This can lead to two major benefits. First, on the SpEC side, we might dump tensor and scalar data into different files. Having two managers offers us more freedom to handle this scenario. Second, having a separate manager for the Klein-Gordon variables alone allows us to construct a standalone scalar CCE system (rather than the current coupled scalar+tensor CCE).

Reply:
Ahh I see. I did not realize this new manager would work in parallel with the old one. I thought it was just a replacement for the KG system. That all makes sense. Thanks!